### PR TITLE
Support for the cluster-api v1beta1 version

### PR DESCRIPTION
**Pull request by iawia002**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Bump the `sigs.k8s.io/cluster-api` package to v1.0.1 to support the `Cluster` v1beta1 version. I have checked the v1beta1 version of the `Cluster` object, and the cluster phase and secret we need have no breaking changes, this patch should work fine.

ref #594

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

I directly used the constants of the `sigs.k8s.io/cluster-api/util/secret` package from cluster-api repo instead of our own definition.

**Does this PR introduce a user-facing change?**:

```release-note
Support for the cluster-api v1beta1 version.
```

/cc @XiShanYongYe-Chang

### CODE REVIEW

File: pkg/clusterdiscovery/clusterapi/clusterapi.go [hunk 1]
```diff
@@ -15,19 +15,25 @@ import (
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/klog/v2"
 	clusterapiv1alpha4 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	clusterapiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	secretutil "sigs.k8s.io/cluster-api/util/secret"
```
Comment (gy95) [inline pkg/clusterdiscovery/clusterapi/clusterapi.go:19]: Does this require us to update vendor versions?

Comment (iawia002) [reply pkg/clusterdiscovery/clusterapi/clusterapi.go:19]: `sigs.k8s.io/cluster-api/api/v1beta1` requires

Comment (gy95) [reply pkg/clusterdiscovery/clusterapi/clusterapi.go:19]: Got it, thanks

**Comment by XiShanYongYe-Chang** [comment]
@iawia002 Thanks, it looks great to me.

**Comment by XiShanYongYe-Chang** [comment]
It works ok on my side. 👍

**Comment by RainbowMango** [comment]
/approve

cc @duguhaotian

**Pull request merged**